odoo/http.py

```diff
@@ -1439,6 +1471,10 @@ def __init__(self, httprequest):
     def _post_init(self):
         self.session, self.db = self._get_session_and_dbname()
 
+        if self.session.created < time.time() - get_session_rotation_interval(self):
+            self.session.created = time.time()
+            self.session.should_rotate = True
```

Comment:
I have mixed feelings about if `_post_init` is the right place for this

Besides this `get_session_rotation_interval` is overly complicated:
- You create a completely new env, new registry, new cursor while you already have everything you need on the request: request.env
- I don't even think we need to make the rotation time configurable, I think we should just decide a good default one and do not make this an option. There is no need to have an option if the default suits everyone. And I think that rotating every 4/8/12 hours is a good protection against session stealing for everyone.

So maybe to keep it kiss I would simply put it in `_save_session`, replacing the condition
```py
        if sess.should_rotate:
            root.session_store.rotate(sess, self.env)  # it saves
```
to
```py
        if sess.should_rotate or sess.create_date < time.time() - SESSION_ROTATION:
            root.session_store.rotate(sess, self.env)  # it saves
```

This `self.session.created = time.time()` I think you can move it inside `def rotate` .
I initially checked if you could use the creation time on the file itself, like we do for the vaccuming based on the modification time
https://github.com/odoo/odoo/blob/a4418749590840d720ff517467142e9c5327474b/odoo/http.py#L933
but looks like
- `os.path.getctime` doesn't work on linux, only linux
- for linux you have `st_birthtime` from `os.stat`, but it looks like it depends on the filesystem format.
So maybe keeping it simple with this create date inside the session is the way to go, But then put it in `def rotate` directly I would say

This is nitpicking but I would rename `created` to `create_date` to keep the same wording as the `create_date` column on regular model. Even though this isn`t a model.

Reply:
_save_session would work too